File: scrimage-filters/src/main/scala/com/sksamuel/scrimage/filter/BackgroundBlendFilter.scala
package com.sksamuel.scrimage.filter

import java.awt.{Color, Graphics2D}

import com.sksamuel.scrimage.{Image, Filter, MutableAwtImage}
import thirdparty.romainguy.BlendComposite._

object BackgroundBlendFilter extends Filter {
  def apply(image: Image) {

    val background = image.fill(new Color(51, 0, 0))
    val g = image.awt.getGraphics.asInstanceOf[Graphics2D]
    g.setComposite(getInstance(BlendingMode.ADD, 1f))
    g.drawImage(background.awt, 0, 0, null)
    g.dispose()
  }
}

File: scrimage-filters/src/main/java/com/sksamuel/scrimage/filter/BackgroundBlendFilter.java
package com.sksamuel.scrimage.filter;

import com.sksamuel.scrimage.ImmutableImage;
import com.sksamuel.scrimage.color.RGBColor;
import thirdparty.romainguy.BlendComposite;
import thirdparty.romainguy.BlendingMode;

import java.awt.*;

public class BackgroundBlendFilter implements Filter {

    @Override
    public void apply(ImmutableImage image) {
        ImmutableImage background = image.fill(new RGBColor(51, 0, 0, 255).toAWT());
        Graphics2D g = (Graphics2D) image.awt().getGraphics();
        g.setComposite(BlendComposite.getInstance(BlendingMode.ADD, 1f));
        g.drawImage(background.awt(), 0, 0, null);
        g.dispose();
    }
}
